### PR TITLE
change auth_code to code

### PR DESCRIPTION
**Pull request by cadecairos**

### CODE REVIEW

**Comment by jbuck** [comment]
Since we need to start doing this, want to add Joi validation of the query parameter for the `code` parameter only?

File: web/server.js [hunk 1]
```diff
@@ -116,6 +117,14 @@ module.exports = function(options) {
       method: 'GET',
       path: '/login/oauth/access_token',
       config: {
+        validate: {
+          query: {
+            code: Joi.string(),
+            client_id: Joi.string(),
+            client_secret: Joi.string(),
+            grant_type: Joi.string()
```
Comment (jbuck) [inline web/server.js]: I think this should be https://github.com/hapijs/joi#anyvalidvalue---aliases-only-equal right?

File: test/web.js [hunk 2]
```diff
@@ -313,4 +313,70 @@ lab.experiment("OAuth", function() {
       });
     });
   });
+
+  lab.test("GET access_token - invalid grant_type", function(done) {
```
Comment (jbuck) [inline test/web.js]: I think these descriptions are wrong

File: test/web.js [hunk 3]
```diff
@@ -313,4 +313,70 @@ lab.experiment("OAuth", function() {
...
+          "Content-Type": "application/x-www-form-urlencoded"
+        }
+      };
+
+      s.inject(authTokenRequest, function(authTokResponse) {
+        var redirectUri = url.parse(authTokResponse.headers.location, true);
+        accessTokenRequest.url += "&code=" + redirectUri.query.code;
+
+        s.inject(accessTokenRequest, function(response) {
+          Code.expect(response.statusCode).to.equal(400);
```
Comment (jbuck) [inline test/web.js]: I think it's also worth checking the error or message properties to make sure the correct error is being triggered

File: test/web.js (hunk 2 above)
Comment (cadecairos) [inline test/web.js]: how?

Comment (jbuck) [inline test/web.js]: blargh, I put this on the wrong line

File: test/web.js [hunk 4]
```diff
@@ -313,4 +313,70 @@ lab.experiment("OAuth", function() {
...
+          done();
+        });
+      });
+    });
+  });
+
+  lab.test("GET access_token - missing client_id", function(done) {
```
Comment (jbuck) [inline test/web.js]: I'm not sure of the value of testing two invalid error conditions here (missing client_id and wrong grant_type)

Comment (cadecairos) [inline test/web.js]: probably a copy paste error on my part

**Comment by jbuck** [comment]
This PR needs to be re-done against develop

**Pull request closed without merging**